### DISCUSSION TITLE
如何在nas中的docker使用呢？（求帮助）

### DISCUSSION THREAD

**Discussion by hychenbin** [docker-easyconnect/docker-easyconnect#57] [Q&A]
没有窗口，如何配置

**Comment by Hagb** [comment]
你的 nas 是什么架构的？（是 x86，还是 x86\_64，还是两者皆不是？）

**Comment by hychenbin** [comment]
x64的，群晖

**Comment by Hagb** [comment]
安装 docker，运行（其中 `7.6.3` 是 EasyConnect 的版本，视你自己的需要可改为 `7.6.7`）：
```
docker pull hagb/docker-easyconnect:7.6.3
```
然后运行容器
```
docker run --device /dev/net/tun --cap-add NET_ADMIN -ti -e PASSWORD=xxxx -v ~/.ecdata:/root -p 5901:5901 -p 1080:1080 hagb/docker-easyconnect
```
可以用 vnc 连 nas 的 `5901` 端口来操作 EasyConnect，代理登陆后 `1080` 端口会开出 socks5

**Comment by hychenbin** [comment]
docker: Error response from daemon: Bind mount failed: '/root/.ecdata' does not exists.
root@DiskStation:~# 
![image](https://user-images.githubusercontent.com/75229976/103170020-e67c1180-487b-11eb-8a01-697ef33e69f1.png)
在容器中的卷中加载了一个目录也不行

**Comment by Hagb** [comment]
> docker: Error response from daemon: Bind mount failed: '/root/.ecdata' does not exists.
> root@DiskStation:~#
> ![image](https://user-images.githubusercontent.com/75229976/103170020-e67c1180-487b-11eb-8a01-697ef33e69f1.png)

可以把 `~/.ecdata` 改为你希望用来存储登录信息的其他目录。

> 在容器中的卷中加载了一个目录也不行

具体是指？

**Comment by hychenbin** [comment]
这个卷就是用实际目录来入登录信息的
另外的端口也开了
![图片](https://user-images.githubusercontent.com/75229976/103183069-478efe00-48eb-11eb-96ed-ca03bb1dfa83.png)

**Comment by Hagb** [comment]
请检查目录`/root/.ecdata`在文件系统中是否存在。看到[一些地方](https://community.home-assistant.io/t/installation-in-a-synology-nas-with-a-docker-apps/2351/20)反映群辉的 docker 是不会自动创建目录的，原本不存在的话需要手动创建。

另外，`-v /root/.ecdata:/root` 这个参数是用于持久化以便下次启动容器后可自动登陆的，并不是必要的。可以先不加这个参数测试一下看看 EasyConnect 的主要功能能不能如你所预期地那样运行。

**Comment by hychenbin** [comment]
正常了，非常感谢，但是联上了，不能访问，如何查看这个服务器后分配的地址呀

**Comment by Hagb** [comment]
> 正常了，非常感谢，但是联上了，不能访问，如何查看这个服务器后分配的地址呀

“联上了”和“不能访问”具体是指？按理来说应该可以以 socks5 的方式连你 nas 的 `1080` 端口来访问 EasyConnect 那边的网络。

**Comment by hychenbin** [comment]
socks5关闭
![图片](https://user-images.githubusercontent.com/75229976/103184831-b02ea880-48f4-11eb-9c28-e0026156f680.png)
运行一分钟后报这个错误

**Comment by Hagb** [comment]
> socks5关闭
> ![图片](https://user-images.githubusercontent.com/75229976/103184831-b02ea880-48f4-11eb-9c28-e0026156f680.png)
> 运行一分钟后报这个错误

我没用过群晖，不过搜索引擎上用“群晖 docker Socket已关闭”作为搜索词查了查，这里的“socket”和“socks5”应该是两回事。

**Comment by Hagb** [comment]
你在 nas 上用 EasyConnect 最终想要达到的效果是怎么样呢？

**Comment by hychenbin** [comment]
加入到单位的网络中去

**Comment by hychenbin** [comment]
单位的服务器已经设好，家里的电脑用客户端正常，手机正常
就是想让这台NAS 也可以加入

**Comment by hychenbin** [comment]
![图片](https://user-images.githubusercontent.com/75229976/103185270-93937000-48f6-11eb-8716-6b980a023285.png)
vnc已经上去，也正常登录上了，但是不通：（

**Comment by Hagb** [comment]
>  vnc已经上去，也正常登录上了，但是不通：（

如果需要给 nas 设全局代理，或许可以参考[我的博客](https://hagb.name/2020/06/26/easyconnect-proxy.html)（不过我不知道有没有什么更好的解决办法），不过还是建议有一定的 GNU/Linux 基础再折腾。

如果只是给 nas 里个别的应用或个别网站设置代理，大概会容易办些。
